huaweicloud/services/dcs/data_source_huaweicloud_dcs_flavors.go

```diff
@@ -0,0 +1,168 @@
+package dcs
+
+import (
+	"github.com/chnsz/golangsdk/openstack/dcs/v2/flavors"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+// DataSourceDcsFlavorsV2 the function is used for data source 'huaweicloud_dcs_flavors'.
+func DataSourceDcsFlavorsV2() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDcsFlavorsV2Read,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Redis", "Memcached",
+				}, false),
+			},
+			"capacity": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
maybe `int` is more friendly.

Reply:
The capacity is actually a float type, but the values of capacity are some specified values, it's better to follow the API.

Reply:
Another reason, when the number of decimal places is uncertain, it is difficult to convert `float64` to `string`.

Reply:
I didn't notice that there have decimals (`0.125`, `0.25`, `0.5`)

Reply:
in the resource.huaweicloud_dcs_instance, the type of capacity is float.

Reply:
modified to float